### PR TITLE
Initial commit

### PR DESCRIPTION
**Pull request by foragerDev**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,54 @@
+dist: bionic
+
+services:
+  - postgresql
+ 
+addons:
+  postgresql: '9.5'
+  apt:
+    packages:
+      - postgresql-9.5-postgis-2.5
```
Comment (umairwaheed) [inline .travis.yml:10]: We don't need PostGis

File: requirements/dev.txt [hunk 2]
```diff
@@ -0,0 +1,13 @@
+Django==3.0.6
+djangorestframework==3.11.0
+django-oauth-toolkit==1.3.2
```
Comment (umairwaheed) [inline requirements/dev.txt:3]: we don't need oauth

File: requirements/dev.txt [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+django-cors-headers==3.3.0
+django-debug-toolbar==2.2
+django-admin-sortable==2.2.3
```
Comment (umairwaheed) [inline requirements/dev.txt:6]: we don't need this

File: scripts/add-aliases [hunk 4]
```diff
@@ -0,0 +1,3 @@
+#!/usr/bin/env bash
```
Comment (umairwaheed) [inline scripts/add-aliases:1]: we don't need this file

File: scripts/lib/lister.py [hunk 5]
```diff
@@ -0,0 +1,135 @@
+import os
```
Comment (umairwaheed) [inline scripts/lib/lister.py:1]: we don't need this file.

File: scripts/lib/utils.py [hunk 6]
```diff
@@ -0,0 +1,22 @@
+# Color codes
```
Comment (umairwaheed) [inline scripts/lib/utils.py:1]: we don't need this file

File: scripts/replacer [hunk 7]
```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env python3
```
Comment (umairwaheed) [inline scripts/replacer:1]: we don't need this file

File: scripts/rm-db [hunk 8]
```diff
@@ -0,0 +1,11 @@
+#!/usr/bin/env bash
```
Comment (umairwaheed) [inline scripts/rm-db:1]: we don't need this file

File: scripts/ship [hunk 9]
```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/env bash
```
Comment (umairwaheed) [inline scripts/ship:1]: we don't need this file

**Comment by umairwaheed** [comment]
@foragerDev this is good scaffolding, but let's make sure that we only add files that we require. https://stackify.com/premature-optimization-evil/

File: .travis.yml [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+  - postgresql
+ 
+addons:
+  postgresql: '9.5'
+  apt:
+    packages:
+      - postgresql-9.5
+
+before_script:
+  - psql -U postgres -c "create user read_user with password 'test'"
```
Comment (umairwaheed) [inline .travis.yml:13]: Why are you doing this?

File: .travis.yml [hunk 11]
```diff
@@ -0,0 +1,50 @@
...
+
+branches:
+  only:
+    - "master"
+    - "production"
```
Comment (umairwaheed) [inline .travis.yml:18]: Why are you adding production and staging?

File: .travis.yml [hunk 12]
```diff
@@ -0,0 +1,50 @@
...
+after_success:
+  - if [ "$TRAVIS_BRANCH" = "master" ]; then coveralls; fi
+
+deploy:
+  - provider: script
+    script: bash scripts/lib/travis-deploy.sh
+    on:
+      branch: "production"
+  - provider: script
+    script: bash scripts/lib/travis-uat-deploy.sh
```
Comment (umairwaheed) [inline .travis.yml:41]: What will this code do?

File: Dockerfile [hunk 13]
```diff
@@ -0,0 +1,36 @@
...
+FROM base as builder
+
+RUN apt update && apt install -y \
+    gcc \
+    git \
+    libpq-dev \
+    postgresql-client \
+    binutils \
+    libproj-dev \
+    gdal-bin
```
Comment (umairwaheed) [inline Dockerfile:12]: You think we need this library?

File: docker-compose.staging.yml [hunk 14]
```diff
@@ -0,0 +1,25 @@
...
+    build:
+      context: .
+      target: prod
+    volumes:
+      - static_volume:/opt/fecundityllc/static
+
+    env_file:
+      ./.env
+
+    container_name: trackit_prod
```
Comment (umairwaheed) [inline docker-compose.staging.yml:19]: This should be trackit_staging

Comment (foragerDev) [reply docker-compose.staging.yml:19]: Ok, I will change it.

File: scripts/lib/travis-deploy.sh [hunk 15]
```diff
@@ -0,0 +1,3 @@
+#!/usr/bin/env bash
```
Comment (umairwaheed) [inline scripts/lib/travis-deploy.sh:1]: There is no need for this file at the moment.

File: Dockerfile (hunk 13 above)
Comment (foragerDev) [reply Dockerfile:12]: OK, I will remove them as well.

File: scripts/lib/travis-uat-deploy.sh [hunk 16]
```diff
@@ -0,0 +1,3 @@
+#!/usr/bin/env bash
```
Comment (umairwaheed) [inline scripts/lib/travis-uat-deploy.sh:1]: There is no need for this file at the moment.

**Comment by umairwaheed** [comment]
@foragerDev we don't need this scripts/lib/git-crypt file. I have added more comments.

File: .dockerignore [hunk 17]
```diff
@@ -0,0 +1,8 @@
+*
+!scripts
+!requirements/dev.txt
+!requirements/prod.txt
+!version.py
```
[the remark this answers is not in the document]
Comment (foragerDev) [reply .dockerignore:5]: I will add.

File: .travis.yml (hunk 11 above)
Comment (foragerDev) [reply .travis.yml:18]: I thought may be necessary.

**Comment by foragerDev** [comment]
I will make the requested changes.

File: .travis.yml (hunk 10 above)
Comment (foragerDev) [reply .travis.yml:13]: > @foragerDev this is good scaffolding, but let's make sure that we only add files that we require. https://stackify.com/premature-optimization-evil/

That makes sense, I am just trying this to learn, how to set up project on docker. And yes there is lesson it and yes I was just spending time that is not actually needed.

**Comment by foragerDev** [comment]
> @foragerDev this is good scaffolding, but let's make sure that we only add files that we require. https://stackify.com/premature-optimization-evil/

Yes this makes sense, I am just trying to learn how to set up project on docker. And there is a lesson in it, probably I was wasting time on something that is not really needed.

**Pull request merged**